src/index.js

```diff
@@ -36,64 +49,118 @@ const Main = ({ ytAPI }) => {
         setInfoText(infoText);
     };
 
+    const handleTokenSubmit = accessToken => {
+        const dbx = new Dropbox({ accessToken, clientId: process.env.REACT_APP_DROPBOX_KEY });
+        const p = initDropboxHelper(dbx);
+        p.then(() => {
+            const p1 = localStorageHelper.syncWithDropbox();
+            p1.then(() => {
+                console.log('setup dropbox complete');
+                setDropboxSetupComplete(true);
+                showInfo('Dropbox setup complete', 2.0);
+            }).catch(error => {
+                console.log('Failed to sync with dropbox - ', error);
+            });
+        }).catch(err => {
+            console.log(err);
+            setDropboxSetupFailed(true);
+            showInfo('Failed to set up Dropbox', 2.0);
+        });
+    };
+
     return (
         <BrowserRouter>
-            <Route
-                path="/"
-                render={({ location }) => (
-                    <>
-                        <Paper elevation={0}>
-                            <Tabs
-                                indicatorColor="primary"
-                                textColor="primary"
-                                value={getTabValue(location.pathname)}
-                                centered
-                            >
-                                <Tab value="editor" label="Editor" component={Link} to="/editor" />
-                                <Tab
-                                    value="notes"
-                                    label="Saved notes"
-                                    component={Link}
-                                    to="/saved_notes"
+            <Switch>
+                <Route
+                    path="/dropbox_oauth"
+                    render={() => {
+                        if (dropboxSetupComplete) {
+                            return <Redirect to="/editor" />;
+                        } else if (dropboxHelper.isInitialized() && !dropboxSetupFailed) {
+                            // Means we have created the api but have not finished setting up the notes folder on dropbox side.
+                            // Disabling handleTokenSubmit.
+                            return <DropboxLogin />;
+                        } else {
```

Comment:
No need for `else`. Just put it in main function. If any of the above conditions are true, the function would have returned early.